src/containers/Ledger/index.js

```diff
@@ -90,36 +90,109 @@ class Ledger extends Component {
     const previousIndex = LedgerIndex - 1
     const nextIndex = LedgerIndex + 1
     const date = new Date(data.close_time)
+    const transactions = data.transactions || []
+    const transactionTypes = {
```

Comment:
Yup, agreed that can be a problem. However, I am not that familiar with the code so you'll have to guide me a bit.

I found this code snippet, and I assume what you mean is we need to a separate field for `TransactionType`:"payment" etc.

```
export interface TransactionMapping {
  Description?: TransactionDescriptionComponent
  Simple: TransactionSimpleComponent
  TableDetail?: TransactionTableDetailComponent
  parser: TransactionParser
  TransactionType: string          //  do this and set in each transaction type? 
}

```

Reply:
Yes, something like that - though I'd recommend calling it something like `TransactionCategory`, since `TransactionType` is usually used for the type of transaction, e.g. Payment vs OfferCreate

Reply:
On it! Also, should I make these two fields optional or just create a component/function and return null in new transaction folders like OfferCancel, NFTokenMint etc.

```
Simple: TransactionSimpleComponent
parser: TransactionParser
```

Reply:
The migration is still ongoing from the old transaction format to the new one. You can leave the original dictionary in for all the transactions that don't exist in this folder yet.

Reply:
So, what I understand is:

Make a folder in Transactions folder -> eg. OfferCreate
```
import { TransactionMapping } from '../types'
import { parser } from './parser'
import { Simple } from './Simple'

export const OfferCancelTransaction: TransactionMapping = {
  Simple,
  TransactionCategory: 'Dex',
  parser,
}

```
and two three files named index, simple and parser.

where simple is:
```
import React from 'react'
import { TransactionSimpleComponent, TransactionSimpleProps } from '../types'

const Simple: TransactionSimpleComponent = (props: TransactionSimpleProps) => {
  const { data } = props

  return <>{data}</>
}
export { Simple }

```

and parser is:

```
export function parser(tx: any, meta: any) {
  return tx
}

```

Reply:
Essentially that is the new setup.  Here is where they are defined and there is a README.

Reply:
Added all the tx types.